src/spriteloader/grf.cpp

```diff
@@ -273,7 +274,10 @@ uint8 LoadSpriteV2(SpriteLoader::Sprite *sprite, uint8 file_slot, size_t file_po
 		byte colour = type & SCC_MASK;
 		byte zoom = FioReadByte();
 
-		if (colour != 0 && (load_32bpp ? colour != SCC_PAL : colour == SCC_PAL) && (sprite_type != ST_MAPGEN ? zoom < lengthof(zoom_lvl_map) : zoom == 0)) {
+		if (colour != 0 &&
+			(load_32bpp ? colour != SCC_PAL : colour == SCC_PAL) &&
+			(sprite_type != ST_MAPGEN ? zoom < lengthof(zoom_lvl_map) && zoom_lvl_map[zoom] >= _settings_client.gui.sprite_zoom_min : zoom == 0)
```

Comment:
please don't do this :P The ternary really got out of hand here :D Either a temporary variable or an extra if or anything, would be greatly appreciated :)

Reply:
Split up and hopefully made a bit clearer what's going on.